src/main/java/net/neoforged/neoforge/transfer/ResourceHandlerUtil.java

```diff
@@ -0,0 +1,582 @@
+     * @param resource    the resource to insert
+     * @param amount      the desired amount of the resource to insert
+     * @param transaction The transaction context for a given insertion.
+     *                    Passing in {@code null} will essentially be the same as doing `execute`,
+     *                    whereas passing in a closeable context allows you to choose if it should be committed.
+     * @return the amount of the resource that was (or would have been, if simulated) inserted
+     */
+    public static <T extends IResource> int insertIndexForced(IResourceHandler<T> handler, T resource, @Range(from = 0, to = Integer.MAX_VALUE) int amount, @Nullable TransactionContext transaction) {
+        try (var subTransaction = Transaction.open(transaction)) {
+            int inserted = 0;
+            int size = handler.size();
+            for (int index = 0; index < size; index++) {
+                inserted += handler.insert(index, resource, amount - inserted, subTransaction);
+                if (inserted >= amount)
+                    return inserted;
+            }
+            subTransaction.commit();
+            return inserted;
+        }
+    }
+
+    /**
+     * Extracts a resource from an {@link IResourceHandler}
+     * Resources will be extracted from the first slot(s) that contain the resource.
+     *
+     * @param <T>         the type of resource handled by the handler
+     * @param handler     the {@link IResourceHandler} to extract the resource from
+     * @param resource    the resource to extract
+     * @param amount      the desired amount of the resource to extract
+     * @param transaction The transaction this transfer is part of, or {@code null} if a transaction should be opened just for this transfer.
+     * @return the amount of the resource that was (or would have been, if simulated) extracted
+     */
+    public static <T extends IResource> int extract(IResourceHandler<T> handler, T resource, @Range(from = 0, to = Integer.MAX_VALUE) int amount, @Nullable TransactionContext transaction) {
+        try (var subTransaction = Transaction.open(transaction)) {
+            int extracted = 0;
+            int size = handler.size();
+            for (int index = 0; index < size; index++) {
+                extracted += handler.extract(index, resource, amount - extracted, subTransaction);
+                if (extracted >= amount)
+                    return extracted;
+            }
+            subTransaction.commit();
+            return extracted;
+        }
+    }
+
+    public static <T extends IResource> int getTotalAmountOf(IResourceHandler<T> handler, T resource) {
+        try (var transaction = Transaction.open(TransactionContext.ROOT)) {
+            //We don't commit allow us to just inquiry the amount
+            return extract(handler, resource, Integer.MAX_VALUE, transaction);
+        }
+    }
+
+    public static boolean any(ItemResource resource) {
```

Comment:
maybe make this `IResource` so that it can be for none `Item` based resources. Also maybe some javadoc to explain its existence which looks to be purely for predicate references. (Why use this over simple lambda `resource->true`?)

```suggestion
    public static boolean any(IResource resource) {
```

Reply:
WOOPS! It was supposed to be! 😂😂😂 Thanks, fixed in current in progress commit :D